src/Parameter.cpp

```diff
@@ -20,6 +20,7 @@ struct ParameterContents {
     std::vector<BufferConstraint> buffer_constraints;
     Expr scalar_default, scalar_min, scalar_max, scalar_estimate;
     const bool is_buffer;
+    bool data_ever_set = false;
```

Comment:
this is the place we realize whenever we add sth to the frontend IR, serialization needs to be updated at the same time

Reply:
I don't understand

Reply:
`data_ever_set` looks like a field that needs to be consistent across serialization (i.e. A parameter that has a data set should also be known as data-ever-set after serialization roundtrip). In this case, we need to add this field to the serialization implementation to properly serialize it.

Reply:
Hmm... didn't think about that. 

I'm actually kinda unsure as to whether we should be saving the scalar value of the Param (if any) to the Serialization or not -- I know we do now (and it's required for the serialization-round-trip-via-JIT hack), but *conceptually*, it seems like the wrong thing to do. 

A Parameter (either scalar or buffer) is conceptually just a placeholder with required-type information. Is it really desirable (in the general sense) to save whatever happens to be stuff in there?

Reply:
it does seem weird to me at first as well, had to admit it.

But now I treat this new field in the PR same/similar to the purpose of `defined`, then it makes more sense